Makefile

```diff
@@ -157,14 +156,36 @@ kind-deploy-controller: generate-operator-yaml install-resources deploy
 .PHONY: deploy-controller
 deploy-controller: kind-deploy-controller
 
+HOSTED ?= none
+
+ifeq ($(HOSTED), hosted)
+KIND_NAMESPACE=cluster2-hosted
+endif
+
 .PHONY: kind-deploy-controller-dev
-kind-deploy-controller-dev: kind-deploy-controller build-images
+kind-deploy-controller-dev: 
+	if [ "$(HOSTED)" = "hosted" ]; then\
+		$(MAKE) kind-deploy-controller-dev-hosted ;\
+	else\
+		$(MAKE) kind-deploy-controller-dev-normal ;\
+		
+	fi;
+
+.PHONY: kind-deploy-controller-dev-normal
+kind-deploy-controller-dev-normal: $(MAKE) kind-deploy-controller
 	@echo Pushing image to KinD cluster
 	kind load docker-image $(REGISTRY)/$(IMG):$(TAG) --name $(KIND_NAME)
 	@echo "Patch deployment image"
 	kubectl patch deployment $(IMG) -n $(KIND_NAMESPACE) -p "{\"spec\":{\"template\":{\"spec\":{\"containers\":[{\"name\":\"$(IMG)\",\"imagePullPolicy\":\"Never\"}]}}}}"
 	kubectl patch deployment $(IMG) -n $(KIND_NAMESPACE) -p "{\"spec\":{\"template\":{\"spec\":{\"containers\":[{\"name\":\"$(IMG)\",\"image\":\"$(REGISTRY)/$(IMG):$(TAG)\"}]}}}}"
 	kubectl rollout status -n $(KIND_NAMESPACE) deployment $(IMG) --timeout=180s
+	@echo Check image change
+	kubectl get deployment -n $(KIND_NAMESPACE) -o yaml config-policy-controller
+
+.PHONY: kind-deploy-controller-dev-hosted
+kind-deploy-controller-dev-hosted:
+	kind load docker-image $(REGISTRY)/$(IMG):$(TAG) --name $(KIND_NAME)
+	sed 's/kindnamespace/$(KIND_NAMESPACE)/g' test/resources/hosted_mode/managed-cluster-addon.yaml | sed 's#img#$(REGISTRY)/$(IMG):$(TAG)#g' | kubectl apply -f- -n cluster2
```

Comment:
Rather than storing the ManagedClusterAddon in this repo, I might prefer that the ManagedClusterAddon is re-annotated, something like using `jq` to set the image tag by updating the JSON:
```
OVERRIDE_ANNOTATION=$(shell kubectl get managedclusteraddon config-policy-controller -o jsonpath='.metadata.annotations["addon.open-cluster-management.io/values"]' | jq -c '.global.imageOverrides.config_policy_controller = "$(REGISTRY)/$(IMG):$(TAG)"')
kubectl annotate --overwrite managedclusteraddon config-policy-controller addon.open-cluster-management.io/values=$(OVERRIDE_ANNOTATION)
```

Reply:
I love this idea Thanks @dhaiducek